test/distributed/test_c10d_nccl.py

```diff
@@ -2685,7 +2685,11 @@ def assert_fut_success(fut):
             work = process_group.allreduce(torch.rand(10).cuda(self.rank))
             work.wait()
             result = work.get_future_result().wait()
-            self.assertEqual(WorkResult(result), WorkResult.COMM_ERROR)
+            self.assertEqual(WorkResult(result), WorkResult.TIMEOUT)
+        else:
+            # other ranks not exiting before rank 0 timeout, this is to avoid
+            # nccl error happening before rank 0 timeouts
+            time.sleep(4)
```

Comment:
nit: can we use variable like `timeout=2` and `sleep(timeout * 2)` to make the test more robust?